### DISCUSSION TITLE
Autoscaling CI - set limit to # of jobs running concurrently

### DISCUSSION THREAD

**Discussion by llifoawing** [dpguthrie/dbtc#60] [Q&A]
The autoscaling CI functionality is a really cool feature and super handy! Wondering if there is a straightforward way to limit the number of jobs that get triggered, as opposed to using the account's max allotted run slots? The project I'm working in has many developers so the CI queues can get pretty long and I don't want to let autoscaling consume 100% of the run slots.

**Comment by dpguthrie** [comment]
I really like the idea of giving the user a bit more flexibility in how things should be queued.  Let me think a bit more on this to see how it could be implemented.  Thanks for raising this though!

**Comment by dpguthrie** [comment]
Let me know if #64 satisfies this ask.

**Reply by llifoawing** [reply]
One scenario I'm not sure is covered is if an autoscaling job limit is set, and due to other jobs running it looks like the max run slots is reached, so autoscaling CI is underutilized.
For example, say the account has 10 run slots, we set the autoscaling limit to 5, there are 3 other non-CI jobs running, so once 2 jobs are triggered by autoscaling CI it will think it has reached its max run slots, but there are enough slots to run another 3 autoscaling CI jobs.
In my case, we define a separate environment for CI, but that might not be generalizable. It's still probably better to potentially underutilize autoscaling CI then have it consume all the run slots.

**Reply by dpguthrie** [reply]
I get what you're after - this should be smarter in that it determines the number of run slots a **CI job** is consuming, not all jobs.  I guess I'm just not convinced that it adds that much more value than just setting a higher threshold.  The way that I'm looking at it is, I want to have at least 2 open run slots for any of my other jobs to consume, so in the scenario above, I would just set `max_run_slots` to 8.  Happy to take another look at this, but it would obviously be a bit more complex.

**Reply by llifoawing** [reply]
Ya I agree. And we shouldn't run into too much non-CI traffic, so I wouldn't expect to run into the above scenario that often. And honestly as long as we're running more than 1 CI job I call that a win 😄 . Thank you!
